### PR TITLE
Update k8s.io/ to newer patch

### PR DESCRIPTION
**Pull request by caseydavenport**
## Description

Fixes https://github.com/projectcalico/calico/issues/7886

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -25,17 +25,19 @@ require (
...
+	k8s.io/apimachinery v0.26.5
+	k8s.io/client-go v0.26.5
+	k8s.io/kube-aggregator v0.26.5
 	k8s.io/utils v0.0.0-20230313181309-38a27ef9d749
 	sigs.k8s.io/controller-runtime v0.14.6
 	sigs.k8s.io/kind v0.17.0 // Do not remove, not used by code but used by build
 	sigs.k8s.io/yaml v1.3.0
 )
 
+require k8s.io/apiserver v0.26.5
```
Comment (tmjd) [inline go.mod:39]: I don't know why we have 2 separate require blocks in this file but do we need a 3rd one with this on its own?

Comment (caseydavenport) [reply go.mod:39]: This was an auto-generated change from `go mod tidy`, not sure why it did this

Comment (caseydavenport) [reply go.mod:39]: I'll try putting it with the others.

Comment (caseydavenport) [reply go.mod]: Fixed

**Review by tmjd — APPROVED** [review]
LGTM

**Pull request merged**